crux_core/src/lib.rs

```diff
@@ -261,3 +291,11 @@ pub struct Request<Effect> {
     pub uuid: Vec<u8>,
     pub effect: Effect,
 }
+
+// This name is too confusing, since we kind of have two "Effects"
+// We've got the `Effect` enum that gets serialised to the shell
+// and the capability level stuff, which this represnets.
+// TODO: Think up a better name...
```

Comment:
Any thoughts on this?

Reply:
I'm not sure I fully underatand what this is - is it the case that the `Effect` defined by the app to enumerate the possible effects the core will emit implements this trait?

Reply:
Does the name `Protocol` resonate? I dunno, the capabilaitiy's protocol with the shell is basically it's tx/rx right? So maybe 🤷

Reply:
Stealing from Smithy, we could call it `Operation` with an associated type called `Result` or `ReturnValue`

Reply:
I was actually thinking `protocol.rs` for the file where we define this stuff in my expanded http capability, so I do quite like it as the name for this trait.  But I'm also happy with `Operation` as the name for a this?  Suppose a protocol _could_ be made up of more than one of these `Operations`, so the `Operation` name makes a bit more sense...